src/extension/configuration.c

```diff
@@ -0,0 +1,258 @@
+            DD_TO_DATADOG_INC));
+#define CALIASES(...) APPLY_N(CALIAS_CHECK_LENGTH, ##__VA_ARGS__)
+#define CALIAS(type, name, default, aliases, ...) aliases
+DD_CONFIGURATION
+#undef CALIAS
+#undef CALIASES
+#undef CONFIG
+
+// Allow for partially defined struct initialization here
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+
+static bool _parse_uint(
+    zai_string_view value, zval *nonnull decoded_value, long long max);
+
+static bool _parse_uint32(
+    zai_string_view value, zval *nonnull decoded_value, bool persistent)
+{
+    UNUSED(persistent);
+    return _parse_uint(value, decoded_value, UINT32_MAX);
+}
+static bool _parse_uint64(
+    zai_string_view value, zval *nonnull decoded_value, bool persistent)
+{
+    UNUSED(persistent);
+    return _parse_uint(value, decoded_value, LONG_MAX);
+}
+
+#define CUSTOM(...) CUSTOM
+// NOLINTNEXTLINE(bugprone-macro-parentheses)
+#define CALIAS_EXPAND(name) {.ptr = name, .len = sizeof(name) - 1},
+#define CALIASES(...)                                                          \
+    ((zai_string_view[]){APPLY_N(CALIAS_EXPAND, ##__VA_ARGS__)})
+#define CONFIG(type, name, ...)                                                \
+    ZAI_CONFIG_ENTRY(DDAPPSEC_CONFIG_##name, name, type, __VA_ARGS__),
+#define CALIAS(type, name, ...)                                                \
+    ZAI_CONFIG_ALIASED_ENTRY(DDAPPSEC_CONFIG_##name, name, type, __VA_ARGS__),
+static zai_config_entry config_entries[] = {DD_CONFIGURATION};
+#undef CALIAS
+#undef CONFIG
+
+bool runtime_config_first_init = false;
+
+static bool _parse_uint(
+    zai_string_view value, zval *nonnull decoded_value, long long max)
+{
+    char *endptr = NULL;
+    const int base = 10;
+    long long ini_value = strtoll(value.ptr, &endptr, base);
+
+    // If there is any error parsing, don't set the value.
+    if (endptr == value.ptr || *endptr != '\0') {
+        return false;
+    }
+
+    if (ini_value < 0) {
+        // If we have a negative value, assume the rate limit is disabled
+        ini_value = 0;
+    } else if (ini_value > max) {
+        ini_value = max;
+    }
+
+    ZVAL_LONG(decoded_value, ini_value);
+    return true;
+}
+
+static char _tolower_ascii(char c)
+{
+    return (char)(c >= 'A' && c <= 'Z' ? c - ('A' - 'a') : c);
+}
+
+static void _copy_tolower(char *restrict dst, const char *restrict src)
+{
+    while (*src) { *(dst++) = _tolower_ascii(*(src++)); }
+}
+
+static void dd_ini_env_to_ini_name(
+    const zai_string_view env_name, zai_config_name *nonnull ini_name)
+{
+    if (env_name.len + DD_TO_DATADOG_INC >= ZAI_CONFIG_NAME_BUFSIZ) {
+        assert(false &&
+               "Expanded env name length is larger than the INI name buffer");
+        return;
+    }
+
+    if (env_name.ptr == strstr(env_name.ptr, "DD_")) {
+        _copy_tolower(ini_name->ptr + DD_TO_DATADOG_INC, env_name.ptr);
+        memcpy(ini_name->ptr, "datadog.", sizeof("datadog.") - 1);
```

Comment:
A bit of a confusing way to write this. I think it would be clearer to do:
```c
memcpy(ini_name->ptr, "datadog.", sizeof("datadog.") - 1);
_copy_tolower(ini_name->ptr + sizeof("datadog.") - 1, env_name.ptr + sizeof("DD_") - 1);
```